notification/logic/PutLogic.java

```diff
@@ -0,0 +1,162 @@
+package com.open.jp.notification.logic;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.time.LocalDateTime;
+
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+import javax.ws.rs.core.Response;
+
+import com.open.jp.notification.common.DataSourceHolder;
+import com.open.jp.notification.errorresponse.ErrorResponse;
+import com.open.jp.notification.request.Request;
+import com.open.jp.notification.request.RequestBody;
+import com.open.jp.notification.response.Responses;
+import com.open.jp.notification.util.StringUtil;
+
+
+@RequestScoped
+public class PutLogic {
+	
+	@Inject
+	private DataSourceHolder ds;
+	
+	private StringUtil stringutil = new StringUtil();
+	
+	public PutLogic() {
+		
+	}
+	
+	public  Response GeneratedErrorRes(ErrorResponseEnum errorresponseenum) {
+	    
+	    ErrorResponse erroresponse = new ErrorResponse();
+	    
+	    erroresponse.setTime(LocalDateTime.now().toString()); 
+	    erroresponse.setCode(errorresponseenum.getErrorStatusCodeMessage()); 
+	    erroresponse.setMessage(errorresponseenum.getErrorMessage()); 
+	    return Response.status(errorresponseenum.getErrorStatus()).entity(erroresponse).build();
+	}
+	
+	public Response service(Request request, RequestBody body, String id) {
+		
+		Responses response = new Responses();
+		
+		StringBuilder userSeqID = new StringBuilder();
+		
+		Response typeresponse = bodyTypeCheck(body);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *typeresponse*. The resource is referenced at line 51. The resource is returned at line 52. There are other execution paths that don&#39;t close the resource or return it, for example, when an else condition at line 51 evaluates to true. Ensure *typeresponse* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).